iai-callgrind-runner/src/runner/tool/format.rs

```diff
@@ -1,37 +1,53 @@
+use std::fmt::Display;
+
+use anyhow::Result;
 use colored::Colorize;
 
-use super::logfile_parser::LogfileSummary;
+use crate::runner::format::format_vertical;
+use crate::runner::summary::ToolRunSummary;
 
-pub struct LogfileSummaryFormatter;
+pub struct ToolRunSummaryFormatter;
 
 pub struct ToolSummaryFormatter;
 
-impl LogfileSummaryFormatter {
+fn print_compare<T: Display>(description: &str, old: Option<T>, new: T) {
+    match old {
+        None => println!("  {description:<18}{}", new.to_string().bold()),
+        Some(old) => println!(
+            "  {description:<18}{:>15}|{old:<15}",
+            new.to_string().bold()
+        ),
+    }
+}
+
+impl ToolRunSummaryFormatter {
     pub fn print(
-        summary: &LogfileSummary,
+        summary: &ToolRunSummary,
         verbose: bool,
         is_multiple: bool,
         force_show_body: bool,
-    ) {
+    ) -> Result<()> {
         if verbose || is_multiple {
-            println!(
-                "  {:<18}{}",
-                "Command:",
-                summary.command.display().to_string().blue().bold()
-            );
-            println!("  {:<18}{}", "PID:", summary.pid.to_string().bold());
+            println!("  {:<18}{}", "Command:", summary.command.blue().bold());
+            print_compare("PID:", summary.old_pid, summary.pid);
 
             if let Some(parent_pid) = summary.parent_pid {
-                println!("  {:<18}{}", "Parent PID:", parent_pid.to_string().bold());
+                print_compare("Parent PID:", summary.old_parent_pid, parent_pid);
             }
         }
 
-        for field in &summary.fields {
+        if let Some(costs) = &summary.costs_summary {
+            print!("{}", format_vertical((None, None), costs.all_diffs())?);
+        }
+
+        for field in &summary.summary {
             println!("  {:<18}{}", format!("{}:", field.0), field.1.bold());
         }
 
-        if (force_show_body || verbose || summary.has_errors()) && !summary.details.is_empty() {
-            let mut iter = summary.details.iter();
+        if (force_show_body || verbose || summary.error_summary.is_some())
```

Comment:
The check for `summary.error_summary.is_some()` isn't enough. You can copy over the `LogfileSummary::has_errors` method to `ToolRunSummary` and use it here.